### PR TITLE
[WIP] Add API documentation

### PR DESCRIPTION
**Pull request by kskyten**
#### Summary:
Improve the documentation of the API.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD3 (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+^^^^^
+.. currentmodule:: elfi.graph
+.. autosummary::
+
+   Graph
+   Node
+   
+.. automodule:: elfi.graph
+   :members:
+
```
Comment (vuolleko) [inline docs/api.rst]: Great. Perhaps this file could be built automatically?

**Comment by vuolleko** [comment]
It would be good to have the list of topics built automatically, otherwise it will be quickly out-of-sync.

**Comment by kskyten** [comment]
I agree that automating this would be ideal. However most of the content is generated directly from the docstrings. I handpicked some classes for the autosummaries and the modules are hardcoded. So what needs to be done manually is adding new/renaming modules and classes. I'll see if there is a way to automate it but on the short/medium term it shouldn't be too much work to do it manually.

**Comment by vuolleko** [comment]
Readthedocs has an advanced build setting that runs setup.py install (which is not currently used/needed), maybe that could be used to run some simple parsing algorithm.

**Comment by vuolleko** [comment]
Closing due to inactivity for now, as this is anyway outdated with the upcoming core.

**Pull request closed without merging**